src/components/markdown-htmlview.component.js

```diff
@@ -0,0 +1,356 @@
+  smartypants: false,
+  // TODO: code highlighting
+  /* highlight: code => {
+    return require('highlight.js').highlightAuto(code).value;
+  }, */
+});
+
+const { height, width } = Dimensions.get('window');
+
+export class MarkdownHtmlview extends Component {
+  props: {
+    source: string,
+    navigation: Object,
+  };
+
+  transformMarkdown = md => {
+    let rendered = marked(md);
+
+    // Transform issues & profiles references into a special markup for HTMLView
+    rendered = rendered.replace(/(\s)#(\d+)/, (match, spacing, number) => {
+      return `${spacing}<issue src="${number}" />`;
+    });
+    rendered = rendered.replace(
+      /(\s)@([_0-9A-Za-z]+)/,
+      (match, spacing, username) => {
+        return `${spacing}<profile src="${username}" />`;
+      }
+    );
+
+    return rendered;
+  };
+
+  navigateToIssue = number => {
+    const params = this.props.navigation.state.params;
+    const issueURL = params.issue
+      ? `${params.issue.repository_url}/issues/${number}`
+      : params.issueURL.replace(/\d+$/, number);
+
+    this.props.navigation.navigate('Issue', {
+      issueURL,
+    });
+  };
+
+  navigateToProfile = username => {
+    this.props.navigation.navigate('Profile', {
+      user: { login: username },
+    });
+  };
+
+  render() {
+    // console.log(this.props.source);
+    const rendererStyles = {
+      newLine: {
+        // backgroundColor: '#00FF00',
+        height: 5,
+        width: 5,
+      },
+      p: {
+        flexDirection: 'row',
+        flexWrap: 'wrap',
+        margin: 0,
+        padding: 0,
+      },
+      li: {
+        flexDirection: 'row',
+        flexWrap: 'wrap',
+        flexGrow: 1,
+        margin: 0,
+        padding: 0,
+        backgroundColor: '#ffcc00',
+      },
+      h1: {
+        text: {
+          fontSize: normalize(24),
+        },
+        wrapper: {
+          borderBottomWidth: 1,
+          marginBottom: 12,
+        },
+      },
+      h2: {
+        text: {
+          fontSize: normalize(20),
+        },
+        wrapper: {
+          borderBottomWidth: 1,
+          marginBottom: 12,
+        },
+      },
+      h3: {
+        text: {
+          fontSize: normalize(18),
+        },
+      },
+      h4: {
+        text: {
+          fontSize: normalize(16),
+        },
+      },
+      h5: {
+        text: {
+          fontSize: normalize(14),
+        },
+      },
+      h6: {
+        text: {
+          fontSize: normalize(12),
+        },
+      },
+    };
+    const myDomElement = (node, index, siblings, parent, defaultRenderer) => {
+      // const onLinkPress = this.props.onLinkPress;
+
+      /* eslint-disable default-case */
+      switch (node.type) {
+        case 'text':
+          if (node.data === '\n') {
+            return <Text style={rendererStyles.newLine} />;
+          }
+
+          /* eslint-disable no-param-reassign */
+          // Convert checkboxes
+          node.data = node.data.replace(
+            /^\s*\[\s?\]/,
```

Comment:
What do you think about moving this regexp into a constant in order to improve its meaning?

Something like `const checkBoxes = /^.....` after doing that you can remove the comment. =]

Reply:
Great readability advice, thanks!